### PR TITLE
Add support for custom FocusNode

### PR DESCRIPTION
**Pull request by marcelgarus**
As proposed in #14.
Please perform some testing before merging into the master branch.

### CODE REVIEW

**Comment by Rahiche** [comment]
CC @marcelgarus  can you take a look?

File: lib/code_input.dart [hunk 1]
```diff
@@ -76,9 +78,18 @@ class CodeInput extends StatefulWidget {
...
-  /// The type of thconstard which shows up.
+  /// A focus node.
   ///
-  /// ## Sample codeconst
+  /// ## Sample code
+  ///
+  /// ```dart
+  /// Code(focusNode: FocusNode())
+  /// ```
+  final FocusNode focusNode;
```
[the remark this answers is not in the document]
Comment (marcelgarus) [reply lib/code_input.dart]: True, I agree this documentation is too short. Do you think we should use a custom example or just use a full-blown comment similar to the `TextField` documentation? Something like this:
```dart
/// Defines the keyboard focus for this widget.
///
/// The [focusNode] is a long-lived object that's typically managed by a
/// [StatefulWidget] parent. See [FocusNode] for more information.
///
/// To give the keyboard focus to this widget, provide a [focusNode] and then
/// use the current [FocusScope] to request the focus:
///
/// ```dart
/// FocusScope.of(context).requestFocus(myFocusNode);
/// ```
///
/// This happens automatically when the widget is tapped.
///
/// To be notified when the widget gains or loses the focus, add a listener
/// to the [focusNode]:
///
/// ```dart
/// focusNode.addListener(() { print(myFocusNode.hasFocus); });
/// ```
///
/// If null, this widget will create its own [FocusNode].
///
/// ## Keyboard
///
/// Requesting the focus will typically cause the keyboard to be shown
/// if it's not showing already.
///
/// On Android, the user can hide the keyboard - without changing the focus -
/// with the system back button. They can restore the keyboard's visibility
/// by tapping on a field.  The user might hide the keyboard and
/// switch to a physical keyboard, or they might just need to get it
/// out of the way for a moment, to expose something it's
/// obscuring. In this case requesting the focus again will not
/// cause the focus to change, and will not make the keyboard visible.
```

[the remark this answers is not in the document]
Comment (Rahiche) [reply lib/code_input.dart]: We can always reference the FocusNode class docs. So a custom example would be better I think

**Comment by Rahiche** [comment]
LGMT, you are good to land it

File: lib/code_input.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (marcelgarus) [reply lib/code_input.dart]: What about something like this?
```dart
/// Defines the keyboard focus for this widget.
///
/// The [focusNode] is a long-lived object that's typically managed by a
/// [StatefulWidget] parent. See [FocusNode] for more information.
///
/// ## Sample code
/// ```dart
/// CodeInput(focusNode: myFocusNode);
/// ```
/// You can then request the `CodeInput`'s focus by calling:
/// ```dart
/// FocusScope.of(context).requestFocus(myFocusNode);
/// ```
```

[the remark this answers is not in the document]
Comment (Rahiche) [reply lib/code_input.dart]: Awesome thanks

**Pull request merged**